protocols/primary-backup/src/main/java/io/atomix/protocols/backup/service/impl/PrimaryBackupServiceContext.java

```diff
@@ -631,6 +631,6 @@ private void changeRole(PrimaryTerm term) {
         future.complete(null);
       }
     });
-    return future;
+    return future.thenRunAsync(() -> threadContext.close());
```

Comment:
isn't this closed in PrimaryBackupServerContext..?

Reply:
oh context, not the factory.

Reply:
yeah it's redundant

Reply:
Actually, no it's not. If a partition shuts down a server without shutting down the `ThreadContextFactory` then the thread context should still be shutdown here.